packages/blade/src/components/Menu/types.ts

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import type { UseFloatingReturn, UseInteractionsReturn, useListItem } from '@floating-ui/react';
+import type React from 'react';
+import { BaseMenuItemProps } from '~components/BaseMenu/BaseMenuItem';
+import type { BoxProps } from '~components/Box';
+import type { FeedbackColors } from '~tokens/theme/theme';
+
+// EXPOSED TYPES
+type MenuProps = {
+  /**
+   * First children is trigger and second children is MenuOverlay
+   **/
+  children: [React.ReactElement, React.ReactElement];
+
+  /**
+   * Open controlled state
+   */
+  isOpen?: boolean;
+
+  /**
+   * On Menu open change callback
+   */
+  onOpenChange?: ({ isOpen }: { isOpen: boolean }) => void;
+
+  /**
+   * Should menu open on click or hover
+   *
+   * @default 'click'
+   */
+  openInteraction?: 'hover' | 'click';
+};
+
+type MenuItemProps = {
+  /**
+   * title of item
+   */
+  title?: string;
+
+  /**
+   * Description text for the item
+   */
+  description?: string;
+
+  /**
+   * Slot to render custom menu items
+   */
+  children?: React.ReactNode;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  as?: BaseMenuItemProps['as'];
+
+  /**
+   * Click handler for MenuItem
+   *
+   * Absense of this prop and href will turn the item into non-interactive item
+   */
+  onClick?: (event: React.MouseEvent) => void;
+
+  /**
+   * Focus handler for MenuItem
+   *
+   * Absense of this prop and href will turn the item into non-interactive item
+   */
+  onFocus?: (event: React.FocusEvent) => void;
+
+  /**
+   * Link to open when item is clicked.
+   *
+   * Absense of this prop and onClick will turn the item into non-interactive item
+   */
+  href?: string;
+
+  /**
+   * HTML target of the link
+   */
+  target?: string;
+
+  /**
+   * Item that goes on left-side of item.
+   *
+   * Will be overriden in multiselect
+   */
+  leading?: React.ReactNode;
+
+  /**
+   * Item that goes on right-side of item.
+   */
+  trailing?: React.ReactNode;
+
+  /**
+   * Item that goes immediately next to the title.
+   */
+  titleSuffix?: React.ReactElement;
+
+  /**
+   * disabled state of item
+   */
+  isDisabled?: boolean;
+
+  /**
+   * Color of item. set to negative for dangerous actions like Delete, Remove, etc
+   */
+  color?: Extract<FeedbackColors, 'negative'>;
```

Comment:
color will always have to be negative with this type.

Reply:
It can be negative or undefined

Reply:
What is undefined mapping to? Should it be neutral or negative in that case then?

Reply:
It maps to one of the surface.text.* color so its not neutral either